GitHubContents.ps1

```diff
@@ -143,3 +153,266 @@ function Get-GitHubContent
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Path
+        The file path for which to set contents.
+
+    .PARAMETER CommitMessage
+        The Git commit message.
+
+    .PARAMETER Content
+        The new file content.
+
+    .PARAMETER BranchName
+        The branch, or defaults to the default branch of not specified.
+
+    .PARAMETER CommitterName
+        The name of the committer of the commit. Defaults to the name of the authenticated user if
+        not specified. If specified, CommiterEmail must also be specified.
+
+    .PARAMETER CommitterEmail
+        The email of the committer of the commit. Defaults to the email of the authenticated user
+        if not specified. If specified, CommitterName must also be specified.
+
+    .PARAMETER AuthorName
+        The name of the author of the commit. Defaults to the name of the authenticated user if
+        not specified. If specified, AuthorEmail must also be specified.
+
+    .PARAMETER AuthorEmail
+        The email of the author of the commit. Defaults to the email of the authenticated user if
+        not specified. If specified, AuthorName must also be specified.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Set-GitHubContent  -Path README.md -OwnerName microsoft -RepositoryName PowerShellForGitHub -CommitMessage 'Adding README.md' -Content '# README' -BranchName master
+
+        Sets the contents of the README.md file on the master branch of the PowerShellForGithub repository.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false,
+        DefaultParameterSetName = 'Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification='One or more parameters (like NoStatus) are only referenced by helper methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    param(
+        [Parameter(
+            Mandatory,
+            Position = 1)]
+        [string] $Path,
+
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [string] $CommitMessage,
+
+        [Parameter(
+            Mandatory,
+            Position = 3)]
+        [string] $Content,
+
+        [Parameter(
```

Comment:
I disagree with the ordering here of the parameters.  I've mentioned this in other PR's today as well.  I feel that the repo being operated on is the source, and should be the initial parameter.  It's consistent with the rest of the repo, and it's consistent with how other built-in PowerShell functions work.